src/datumaro/components/media.py

```diff
@@ -355,7 +355,11 @@ def save(
         ext: Optional[str] = None,
         crypter: Crypter = NULL_CRYPTER,
     ):
-        data = self.data
+        if callable(self.data):
+            data = np.array(self.data())
+        else:
+            data = self.data
```

Comment:
Does `ImageFromData` already have this logic as follow (inherited by `FromDataMixin`)?

https://github.com/openvinotoolkit/datumaro/blob/ea19ad2f3801a10b033e30e70c74215333eeb34a/src/datumaro/components/media.py#L179-L183

Reply:
Thank you for your help. I updated the mock video frame generation with propertymock.